### PR TITLE
 Add status section under the MachineHealthCheck object

### PR DESCRIPTION
**Pull request by cynepco3hahue**
MHC status section will include all nodes with unhealthy conditions targeted by this MHC object, all targeted unhealthy conditions and the total number of healthy machines.

Example of status section
```
Status:
  Targeted Conditions:
    Name:    Ready
    Status:  Unknown
    Name:    Ready
    Status:  False
  Targeted Machines:
    Healthy:  false
    Name:     test-1-qvx66-worker-0-224x4
    Unhealthy Conditions:
  Total Healthy Machines:  1
```

### CODE REVIEW

File: install/0000_30_machine-api-operator_02_machine.crd.yaml [hunk 1]
```diff
@@ -6,31 +6,6 @@ metadata:
     controller-tools.k8s.io: "1.0"
   name: machines.machine.openshift.io
 spec:
```
Comment (enxebre) [inline install/0000_30_machine-api-operator_02_machine.crd.yaml]: why is this being removed?

File: install/0000_30_machine-api-operator_03_machineset.crd.yaml [hunk 2]
```diff
@@ -6,27 +6,6 @@ metadata:
     controller-tools.k8s.io: "1.0"
   name: machinesets.machine.openshift.io
 spec:
-  additionalPrinterColumns:
```
Comment (enxebre) [inline install/0000_30_machine-api-operator_03_machineset.crd.yaml]: why is this being removed?

**Comment by enxebre** [comment]
only last two commits belong to this PR

File: install/0000_30_machine-api-operator_02_machine.crd.yaml (hunk 1 above)
Comment (cynepco3hahue) [reply install/0000_30_machine-api-operator_02_machine.crd.yaml]: looks like it is the result of the `make gen-crd`, need to check why it happens, probably it was some change under the cluster-api repository

File: install/0000_30_machine-api-operator_03_machineset.crd.yaml (hunk 2 above)
Comment (cynepco3hahue) [reply install/0000_30_machine-api-operator_03_machineset.crd.yaml]: the same

**Comment by cynepco3hahue** [comment]
> only last two commits belong to this PR

@enxebre yep I based this PR on top of https://github.com/openshift/machine-api-operator/pull/327 and I will create an additional two PR's.
Can you please review https://github.com/openshift/machine-api-operator/pull/327?

**Comment by vikaschoudhary16** [comment]
Please rebase.

**Comment by cynepco3hahue** [comment]
We will need to decide what fields do we want to expose under the status, will open another PR after it.

**Pull request closed without merging**